services/httpd/handler.go

```diff
@@ -1595,6 +1595,11 @@ func authenticate(inner func(http.ResponseWriter, *http.Request, meta.User), h *
 					return
 				}
 			case BearerAuthentication:
+				if h.Config.SharedSecret == "" {
+					atomic.AddInt64(&h.stats.AuthenticationFailures, 1)
+					h.httpError(w, "bearer auth disabled", http.StatusUnauthorized)
```

Comment:
Shouldn't this message indicate why the auth failed, i.e., that the provided shared secret was empty?

Reply:
I debated what message to return to the client and decided less info is better for security. Thought about logging it but that would allow a DoS by filling the log. Maybe I should log it using `sync.Once`?

Reply:
Maybe I'm confused. Isn't this the case where the user has setup bearer auth but they have forgotten to fill out a config value?

Reply:
There are other existing ways that a malicious unauthenticated user can spam the server logs, especially if HTTP logging is enabled.

I'm fine with using sync.Once here, but I'm also fine looking for a pattern to spread widely on those other cases.

Reply:
There isn't a separate config setting for enabling bearer auth. If the shared secret is set, it is enabled. If it's not, bearer auth is disabled.